### PR TITLE
Feat/bundle ref map

### PR DESCRIPTION
**Pull request by LuisTejedaS**
## Description

Add reference map to bundle output.

The reference map will be generated when the option "includeReferenceMap" is set to true.

the ref map is generated with the xpath of the resolved elements and the file path of the file they were taken from

## How to test this ticket
 run tests

## Sample Input and Output (remove if not needed)

```
 '//definitions//types//xsd:schema[1]//xsd:element[@name="ElementType"]' :
    {
        path: '/Types.xsd',
        type: 'inline'
      });
```

### Checklist:

- [X] Checked all the config values
- [X] Added relevant unit tests for my code
- [ ] Did not add relevant unit test for my code, by created {{ticket#}} to
      handle in the future

### CODE REVIEW

**Review by Carlos-Veloz — APPROVED** [review]

File: lib/utils/WSDLMerger.js [hunk 1]
```diff
@@ -596,36 +730,118 @@ class WSDLMerger {
...
    * @param {object} rootPrincipalPrefix the principal prefix for this root
    * @param {string} wsdlRoot the root Tag from this root object
-   * @param {object} importedTag the tag object that is being imported
+   * @param {object} importedTagInformation the tag object that is being imported with all the information
+   * @param {string} schemaPrefix the prefix of the schema
    * @returns {object} undefined
    */
-  assignImportedPropertiesSchemaNotTypesTag(root, rootPrincipalPrefix, wsdlRoot, importedTag) {
+  assignImportedPropertiesSchemaNotTypesTag(root, rootPrincipalPrefix, wsdlRoot, importedTagInformation) {
     root[rootPrincipalPrefix + wsdlRoot][rootPrincipalPrefix + ATTRIBUTE_TYPES] = [];
```
Comment (erfemega) [inline lib/utils/WSDLMerger.js:754]: Here

File: lib/utils/WSDLMerger.js [hunk 2]
```diff
@@ -164,20 +179,22 @@ class WSDLMerger {
   /**
    * Merge different files to a single WSDL
    * @param {object} parsedData the parsedData used to bundle the files
-   ** @param {array} parsedData.parsedXMLsAndFileNames  An object with the parsed input xmls with filenames
-   ** @param {array} parsedData.parsedXMLs An array of all the parsed xml files from input
-   ** @param {array} parsedData.parsedWSDL An array of all the parsed wsdl files detected in input
-   ** @param {array} parsedData.rootFileNames An array of all the file names from the detected root files
-   ** @param {array} parsedData.rootFiles An array with all the detected root files content
+   * @param {array} parsedData.parsedXMLsAndFileNames  An object with the parsed input xmls with filenames
```
[the remark this answers is not in the document]
Comment (LuisTejedaS) [reply lib/utils/WSDLMerger.js:182]: Yes, did not know it was for it

File: lib/utils/WSDLMerger.js [hunk 3]
```diff
@@ -596,36 +730,118 @@ class WSDLMerger {
...
+  }
+
+  /**
+   * Assigns the schema when there are not types in to root
    * @param {object} root the root file parsed
    * @param {object} rootPrincipalPrefix the principal prefix for this root
    * @param {string} wsdlRoot the root Tag from this root object
-   * @param {object} importedTag the tag object that is being imported
+   * @param {object} importedTagInformation the tag object that is being imported with all the information
+   * @param {string} schemaPrefix the prefix of the schema
```
[the remark this answers is not in the document]
Comment (LuisTejedaS) [reply lib/utils/WSDLMerger.js:750]: Removed it from the jsDoc

File: lib/utils/WSDLMerger.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (LuisTejedaS) [reply lib/utils/WSDLMerger.js]: Added again

**Pull request merged**